lib/api/bucketGetTagging.js

```diff
@@ -0,0 +1,106 @@
+
+//  Sample XML response:
+/*
+    <Tagging>
+        <TagSet>
+            <Tag>
+                <Key>string</Key>
+                <Value>string</Value>
+            </Tag>
+            <Tag>
+                <Key>string</Key>
+                <Value>string</Value>
+            </Tag>
+        </TagSet>
+    </Tagging>
+*/
+
+/**
+ * Convert Versioning Configuration object of a bucket into xml format.
+ * @param {array} tags - set of bucket tags
+ * @return {string} - the converted xml string of the versioning configuration
+ */
+function convertToXml(tags) {
+    const xml = [];
+
+    xml.push('<?xml version="1.0" encoding="UTF-8" standalone="yes"?>',
+        '<Tagging>',
+        '<TagSet>'
+    );
+
+    tags.forEach(tag => {
+        xml.push('<Tag>');
+        xml.push(`<Key>${escapeForXml(tag.Key)}</Key>`);
+        xml.push(`<Value>${escapeForXml(tag.Value)}</Value>`);
+        xml.push('</Tag>');
+    });
+
+    xml.push('</TagSet>',
+        '</Tagging>');
+
+    return xml.join('');
+}
+
+/**
+ * bucketGetVersioning - Return Versioning Configuration for bucket
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to respond to http request
+ *  with either error code or xml response body
+ * @return {undefined}
+ */
+function bucketGetTagging(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketGetTagging' });
+
+    const { bucketName, headers } = request;
+
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketGetTagging',
+        request,
+    };
+
+    let bucket = null;
+    return waterfall([
+        next => metadataValidateBucket(metadataValParams, log,
+            (err, b) => {
+                bucket = b;
+                return next(err);
+            }),
+        next => checkExpectedBucketOwner(headers, bucket, log, next),
+        next => {
+            const tags = bucket.getTags();
+            if (!tags || !tags.length) {
+                log.debug('bucket TagSet does not exist', {
+                    method: 'bucketGetTagging',
+                });
+                return next(errors.NoSuchTagSet);
+            }
+            const xml = convertToXml(tags);
+            pushMetric('getBucketTagging', log, {
+                authInfo,
+                bucket: bucketName,
+            });
+            const corsHeaders = collectCorsHeaders(request.headers.origin,
+                request.method, bucket);
+            return callback(null, xml, corsHeaders);
+        },
+    ], (err) => {
+        const corsHeaders = collectCorsHeaders(request.headers.origin,
+            request.method, bucket);
+        log.debug('error processing request',
+            { method: 'bucketGetTagging', error: err });
+        return callback(err, corsHeaders);
```

Comment:
Is it expected to always log an error?
```suggestion
        const corsHeaders = collectCorsHeaders(request.headers.origin,
            request.method, bucket);
        if (err) {
            log.debug('error processing request',
                { method: 'bucketGetTagging', error: err });
        }
        return callback(err, corsHeaders);
```

Reply:
Oops I misread my work on this one, fixed c196c6db3ba238217796076650ef9ab23984dfb2